File: backend/src/models/Product.js
import mongoose from 'mongoose';

const productSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String },
  price: { type: Number, required: true },
  images: [{ type: String }],
  category: { type: String, index: true },
  stock: { type: Number, default: 0 },
  rating: { type: Number, default: 0 }
}, { timestamps: true });

export default mongoose.model('Product', productSchema);

File: flutter_app/lib/models/product.dart
class Product {
  final String id;
  final String title;
  final String? description;
  final double price;
  final String? image;
  final String? category;

  Product({required this.id, required this.title, this.description, required this.price, this.image, this.category});

  factory Product.fromJson(Map<String, dynamic> j) => Product(
    id: j['_id'],
    title: j['title'],
    description: j['description'],
    price: (j['price'] as num).toDouble(),
    image: (j['images'] != null && j['images'].isNotEmpty) ? j['images'][0] : null,
    category: j['category'],
  );
}
